R/LearnerSurvCTree.R

```diff
@@ -0,0 +1,115 @@
+            tags = "train"),
+          ParamDbl$new("logmincriterion", tags = "train"),
+          ParamInt$new("minsplit", lower = 1L, default = 20L, tags = "train"),
+          ParamInt$new("minbucket", lower = 1L, default = 7L, tags = "train"),
+          ParamDbl$new("minprob", lower = 0, default = 0.01, tags = "train"),
+          ParamLgl$new("stump", default = FALSE, tags = "train"),
+          ParamLgl$new("lookahead", default = FALSE, tags = "train"),
+          ParamLgl$new("MIA", default = FALSE, tags = "train"),
+          ParamInt$new("nresample", lower = 1L, default = 9999L,
+            tags = "train"),
+          ParamDbl$new("tol", lower = 0, tags = "train"),
+          ParamInt$new("maxsurrogate", lower = 0L, default = 0L,
+            tags = "train"),
+          ParamLgl$new("numsurrogate", default = FALSE, tags = "train"),
+          ParamInt$new("mtry", lower = 0L, special_vals = list(Inf),
+            default = Inf, tags = "train"),
+          ParamInt$new("maxdepth", lower = 0L, special_vals = list(Inf),
+            default = Inf, tags = "train"),
+          ParamLgl$new("multiway", default = FALSE, tags = "train"),
+          ParamInt$new("splittry", lower = 0L, default = 2L, tags = "train"),
+          ParamLgl$new("intersplit", default = FALSE, tags = "train"),
+          ParamLgl$new("majority", default = FALSE, tags = "train"),
+          ParamLgl$new("caseweights", default = FALSE, tags = "train"),
+          ParamUty$new("applyfun", tags = "train"),
+          ParamInt$new("cores", special_vals = list(NULL), default = NULL,
+            tags = "train"),
+          ParamLgl$new("saveinfo", default = TRUE, tags = "train"),
+          ParamLgl$new("update", default = FALSE, tags = "train"),
+          ParamFct$new("splitflavour", default = "ctree",
+            levels = c("ctree", "exhaustive"), tags = c("train", "control")),
+          ParamUty$new("offset", tags = "train"),
+          ParamUty$new("cluster", tags = "train"),
+          ParamUty$new("scores", tags = "train"),
+          ParamLgl$new("doFit", default = TRUE, tags = "train"),
+          ParamUty$new("pargs", default = mvtnorm::GenzBretz, tags = "train")
+        )
+      )
+      ps$add_dep("nresample", "testtype", CondEqual$new("MonteCarlo"))
+
+      super$initialize(
+        id            = "surv.ctree",
+        packages      = c("partykit", "coin", "distr6"),
+        feature_types = c("integer", "numeric", "factor", "ordered"),
+        predict_types = c("distr", "crank"),
+        param_set     = ps,
+        properties    = "weights"
+      )
+    }
+  ),
+
+  private = list(
+    .train = function(task) {
+      pars = self$param_set$get_values(tags = "train")
+
+      if ("weights" %in% task$properties) {
+        pars$weights = task$weights$weight
+      }
+
+      mod <- mlr3misc::invoke(partykit::ctree, formula = task$formula(),
+        data = task$data(), .args = pars)
```

Comment:
It's safe to delete "mod <- " here

Reply:
true, thks. Forgot to remove after testing.